### PR TITLE
Change ConfigError for an enum

### PR DESCRIPTION
**Pull request by edouardparis**
It makes easier to match errors
while reading configuration file
for external module.

### CODE REVIEW

**Review by darosior — COMMENTED** [review]
Concept ACK - been wanting that :)

**Comment by danielabrozzoni** [comment]
utACK ad371cc60b03428caba21c15ded5849e2e6a5d0a

**Comment by darosior** [comment]
I'd like to get #344 in first, as this one would be much easier to rebase :)

File: src/revaultd.rs [hunk 1]
```diff
@@ -358,23 +358,38 @@ pub struct RevaultD {
...
         fs::create_dir_all(datadir_path)
+            .map_err(|e| CreateDatadirError(datadir_path.to_path_buf(), e.to_string()))
     };
 }
 
+#[derive(Debug)]
+pub struct CreateDatadirError(PathBuf, String);
+impl fmt::Display for CreateDatadirError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Could not create datadir: '{:?}: {}", self.0, self.1)
```
Comment (darosior) [inline src/revaultd.rs:387]: nit: missing an unquote

**Review by darosior — COMMENTED** [review]
ACK 78defb176afce53071e2ed3d7d2bc8fa7de9ed99 this is so much better

**Comment by darosior** [comment]
Arg, again an instance of the coordinator bug

**Comment by darosior** [comment]
Again the coordinator bug! This shouldn't hold up this PR.

**Review by darosior — COMMENTED** [review]
light review ACK 8a8e56fd3626acd648748915fe50b4040af13386

**Pull request merged**